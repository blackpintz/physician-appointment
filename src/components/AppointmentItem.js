import React from 'react';
import propTypes from 'prop-types';

const AppointmentItem = ({ item }) => {
  const { city, user, physician } = item;
  const { email } = user;
  const { category } = physician;
  return (
    <>
      <p>Each Appointment goes here.</p>
      <p>{`Email: ${email}`}</p>
      <p>{`Category: ${category}`}</p>
      <p>{city}</p>
    </>
  );
};

AppointmentItem.propTypes = {
  item: propTypes.objectOf(propTypes.oneOfType([
    propTypes.string,
    propTypes.number,
    propTypes.object,
  ])),
};

AppointmentItem.defaultProps = {
  item: { city: 'no value' },
};

export default AppointmentItem;
