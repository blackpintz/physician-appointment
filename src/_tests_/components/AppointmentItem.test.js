import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import AppointmentItem from '../../components/AppointmentItem';
import appointments from '../fixtures/appointments';

Enzyme.configure({ adapter: new Adapter() });

test('should render AppointmentItem component', () => {
  const wrapper = shallow(<AppointmentItem item={appointments[0]} />);
  expect(wrapper).toMatchSnapshot();
});
