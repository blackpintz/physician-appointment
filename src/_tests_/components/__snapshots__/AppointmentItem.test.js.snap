// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render AppointmentItem component 1`] = `
<Card
  body={false}
  className="w-50 mb-3"
  key="1"
>
  <CardHeader
    as="h5"
  />
  <CardBody>
    <CardTitle
      className="text-secondary"
    >
      Eye Specialist appointment on 2021-02-07, at Mwingi
    </CardTitle>
  </CardBody>
</Card>
`;
